move/example/sources/its/its.move

```diff
@@ -0,0 +1,177 @@
+module example::its {
+    use std::ascii;
+    use std::ascii::{String};
+    use std::type_name;
+
+    use sui::event;
+    use sui::address;
+    use sui::hex;
+    use sui::coin::{Coin};
+    use sui::sui::SUI;
+    use sui::clock::Clock;
+
+    use axelar_gateway::channel::{Self, Channel, ApprovedMessage};
+    use axelar_gateway::discovery::{Self, RelayerDiscovery, Transaction};
+
+    use gas_service::gas_service::GasService;
+
+    use its::service;
+    use its::its::ITS;
+    use its::token_id::TokenId;
+
+    public struct Singleton has key {
+        id: UID,
+        channel: Channel,
+    }
+
+    public struct Executed has copy, drop {
+        source_chain: String,
+        source_address: vector<u8>,
+        data: vector<u8>,
+        amount: u64,
+    }
+
+    fun init(ctx: &mut TxContext) {
+        let singletonId = object::new(ctx);
+        let channel = channel::new(ctx);
+        transfer::share_object(Singleton {
+            id: singletonId,
+            channel,
+        });
+    }
+
+    public fun register_transaction(discovery: &mut RelayerDiscovery, singleton: &Singleton, its: &ITS) {
+        let mut arguments = vector::empty<vector<u8>>();
+
+        // Singleton object
+        let mut arg = vector::singleton<u8>(0);
+        arg.append(object::id_address(singleton).to_bytes());
+        arguments.push_back(arg);
+
+        // ITS object
+        arg = vector::singleton<u8>(0);
+        arg.append(object::id_address(its).to_bytes());
+        arguments.push_back(arg);
+
+        // payload
+        arg = vector[ 3 ];
+        arguments.push_back(arg);
```

Comment:
create a helper function to construct each argument instead? 

something like this
```move
fun create_argument<T>(prefix: u8, value: T): vector<u8> {
   let mut arg = vector::singleton<u8>(prefix);
   arg.append(object::id_address(value).to_bytes());
   arg 
}
```

then, maybe reuse it like below

```
let arguments = vector[
  create_argument(0, singleton),
  create_argument(0, its),
  vector[3],
];
```

and we can reuse it in the `get_transaction` function as well.